### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'get')

### ISSUE THREAD

**Issue by wadeww** [dataform-co/dataform#1642]
I'm running into a compilation error with both the `@dataform/cli` and the Dataform editor in GCP when using `resolve()` for tables in a specific BigQuery dataset. Other datasets work fine.

The full stack trace when using the cli is:

```
TypeError: Cannot read properties of undefined (reading 'get')
    at ActionIndex.find (node_modules/@dataform/core/bundle.js:24301:17)
    at Session.resolve (node_modules/@dataform/core/bundle.js:23855:49)
    at TableContext.resolve (node_modules/@dataform/core/bundle.js:22640:35)
    at Object.sqlContextable (definitions/picos/display_brand_compliance.sqlx:32:12)
    at node_modules/@dataform/core/bundle.js:23813:49
    at TableContext.apply (node_modules/@dataform/core/bundle.js:22686:20)
    at Table.compile (node_modules/@dataform/core/bundle.js:22576:36)
    at node_modules/@dataform/core/bundle.js:23994:46
    at Array.forEach (<anonymous>)
    at Session.compileGraphChunk (node_modules/@dataform/core/bundle.js:23992:17)
```

I'm trying to write unit tests for my query, so I want to use `resolve()`. Most of the tables I need to reference work fine, but all of the tables within a specific dataset (`datastream_append_public`), as well as tables referenced from another GCP project throw this error when compiling. When referenced with backticks, it compiles without error:

Simple `.sqlx` definitions for reference

## Good Dataset
```sql
config {
  type: "table",
  schema: "test",
  name: "test_table",
}

SELECT * FROM ${resolve("datastream_syncpublic", "priority")} -- compiles
-- SELECT * FROM ${resolve("customer_data_sync", "authorized_materials")} -- compiles
```

## Problematic Dataset
```sql
config {
  type: "table",
  schema: "test",
  name: "test_table",
}


SELECT * FROM `datastream_append_public.inventory_inventoryitem` -- compiles
-- SELECT * FROM `datastream_append_public.audit_audits` -- compiles

-- SELECT * FROM ${resolve("datastream_append_public", "inventory_inventoryitem")} -- error
-- SELECT * FROM ${resolve("datastream_append_public", "audit_audits")} --error
```

I'm trying to figure out what the difference with the failing dataset is. All of them are in the same region. `datastream_append_public` and `datastream_syncpublic` are both datasets that are replicated to BigQuery with Datastream from Postgres tables.

I'm not sure where to look next to troubleshoot, so any help would be appreciated. Thanks!

**Comment by BenBirt** [comment]
I'm pretty sure this is happening because Dataform has no actions registered with the schema `datastream_append_public`. (I'm fixing this to have a better error message in https://github.com/dataform-co/dataform/pull/1643.)

I don't know where `datastream_append_public.inventory_inventoryitem` or `datastream_append_public.audit_audits` are defined, but note that `ref()`/`resolve()` can only resolve actions which are _in the Dataform compilation graph_. I suspect that these are tables that exist in BQ but not in Dataform (?), in which case the right thing to do is either refer to them directly, or use a `declaration` (useful for unit testing your queries, if you so desire).

If I'm wrong - and you think that these actions are part of the Dataform compilation graph - please do include the code that defines them and we can look more deeply.

**Comment by BenBirt** [comment]
Released in `2.8.4`: https://github.com/dataform-co/dataform/releases/tag/2.8.4

**Comment by wadeww** [comment]
That's exactly what it was. I didn't realize I had declarations defined for the other datasets that were working.

I added a `declaration` definition for `gnarnia_datastream_append_public` and it resolved the issue. Thank you!
